lib/matplotlib/cm.py

```diff
@@ -275,6 +275,17 @@ def to_rgba(self, x, alpha=None, bytes=False, norm=True):
         x = ma.asarray(x)
         if norm:
             x = self.norm(x)
+
+        try:
+            np_alpha = ma.asarray(alpha)
+            if np_alpha.ndim == 1 and np_alpha.shape == x.shape:
```

Comment:
To clarify, the case of higher dimensional would be supplying a 2 or 3 dimensional `x` and a 2 dimensional `alpha` array?